extension/src/setup/util.ts

```diff
@@ -25,3 +25,20 @@ export const showSetupOnFirstUse = async (
     void setUserConfigValue(ConfigKey.DO_NOT_SHOW_SETUP_AFTER_INSTALL, true)
   }
 }
+
+export const showSetupOnFirstUse = (
+  globalState: Memento,
+  workspaceState: Memento
+): Promise<void> | undefined => {
+  const installed = globalState.get(GlobalPersistenceKey.INSTALLED, false)
+
+  if (installed) {
+    return
+  }
+
+  void globalState.update(GlobalPersistenceKey.INSTALLED, true)
+
+  // old users won't have the installedKey even if it's not a new install
+  const workspaceKeys = workspaceState.keys()
+  return showSetupToast(workspaceKeys.length === 0)
```

Comment:
We check for workspaceKeys length since an old user will have these set. I suppose we could take off this extra check in about a month like we've done in the past.

Reply:
[Q] Same question. Are you sure everything in the workspaceState only relates to our extension?

Reply:
I believe that both workspaceState and globalState both relate only to our extension since looking at the docs for `ExtensionContext`:

```
     * An extension context is a collection of utilities private to an
     * extension.
```

Plus I was console logging both states before I touched anything. `globalState` was completely empty and `workspaceState` only had keys set by our extension.